share/revng/cmake/TupleTreeGenerator.cmake

```diff
@@ -147,40 +141,29 @@ function(tuple_tree_generator_impl)
             ${EXTRA_TARGETS} generate-node_modules)
 endfunction()
 
-# Extracts tuple_tree_generator YAML definitions from the given header files
-#
-function(tuple_tree_generator_extract_definitions_from_headers HEADERS
-         OUTPUT_FILE)
-  add_custom_command(
-    OUTPUT "${OUTPUT_FILE}"
-    COMMAND "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/extract_yaml.py"
-            --output "${OUTPUT_FILE}" TUPLE-TREE-YAML ${HEADERS}
-    DEPENDS ${HEADERS})
-endfunction()
-
 # Computes the list of headers and C++ source files that will be generated by
 # tuple_tree_generator. Note: the output variables will be overwritten
-function(tuple_tree_generator_compute_generated_cpp_files SOURCE_HEADERS
+function(tuple_tree_generator_compute_generated_cpp_files SCHEMA_PATH
          HEADERS_DIR GENERATED_HEADERS_VARIABLE GENERATED_IMPLS_VARIABLE)
-  # Empty output variables
-  set(LOCAL_GENERATED_HEADERS_VARIABLE "${HEADERS_DIR}/ForwardDecls.h")
-  set(LOCAL_GENERATED_IMPLS_VARIABLE)
-
-  foreach(HEADER ${SOURCE_HEADERS})
-    # TODO: we should not be generating /Impl/*.cpp files in an include/*
-    # directory
-    #
-    # TODO: this piece of code is tightly coupled with cppheaders.py
-    get_filename_component(HEADER_FILENAME "${HEADER}" NAME)
-    get_filename_component(HEADER_FILENAME_WE "${HEADER}" NAME_WE)
-    set(EARLY_OUTPUT "${HEADERS_DIR}/Early/${HEADER_FILENAME}")
-    set(LATE_OUTPUT "${HEADERS_DIR}/Late/${HEADER_FILENAME}")
-    list(APPEND LOCAL_GENERATED_HEADERS_VARIABLE "${EARLY_OUTPUT}")
-    list(APPEND LOCAL_GENERATED_HEADERS_VARIABLE "${LATE_OUTPUT}")
-  endforeach()
+  execute_process(
+    COMMAND "${SCRIPTS_ROOT_DIR}/tuple-tree-generate-cpp-paths.py"
+            "${SCHEMA_PATH}" "${HEADERS_DIR}"
+    RESULT_VARIABLE GENERATE_PATHS_EXIT_CODE
+    OUTPUT_VARIABLE GENERATED_PATHS
+    ERROR_VARIABLE GENERATE_PATHS_STDERR)
+
+  if(NOT "${GENERATE_PATHS_EXIT_CODE}" EQUAL "0")
+    message(
+      FATAL_ERROR
+        "the script for computing the list of generated paths has failed." "\n"
+        "${GENERATE_PATHS_STDERR}")
+  endif()
```

Comment:
Is this needed? Doesn't `execute_process` already throw an error if the exit code is non-zero?

Reply:
The CMake documentation for [`execute_process`](https://cmake.org/cmake/help/latest/command/execute_process.html) doesn't mention this, so I assume it doesn't.

Reply:
Ok, you're right, drop the `if` and just use `COMMAND_ERROR_IS_FATAL ANY`